core/src/main/scala/io/projectglow/sql/expressions/LinearRegressionExpr.scala

```diff
@@ -16,19 +16,36 @@
 
 package io.projectglow.sql.expressions
 
+import org.apache.spark.TaskContext
 import org.apache.spark.sql.SQLUtils
 import org.apache.spark.sql.catalyst.InternalRow
-import org.apache.spark.sql.catalyst.expressions.codegen.CodegenFallback
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, CodegenFallback, ExprCode}
 import org.apache.spark.sql.catalyst.expressions.{Expression, ImplicitCastInputTypes, TernaryExpression}
 import org.apache.spark.sql.catalyst.util.ArrayData
 import org.apache.spark.sql.types._
 
+object LinearRegressionExpr {
+  val matrixUDT = SQLUtils.newMatrixUDT()
+  val state = new ThreadLocal[CovariateQRContext]
```

Comment:
Can we make an abstraction for how we get/set/remove state? The general logic seems to match between logit and linreg.

Reply:
I tried to do that, but I don't think there's actually much to share other than registering the callback.